analytic_engine/src/memtable/key.rs

```diff
@@ -208,6 +209,23 @@ pub fn internal_key_for_seek<'a>(
     Ok(&scratch[..])
 }
 
+/// Encode internal key from user key for seek
+///
+/// - user_key: the user key to encode
+/// - sequence: the sequence number to encode into internal key
+/// - scratch: buffer to store the encoded internal key, the scratch will be
+///   clear
+///
+/// Returns the slice to the encoded internal key
+pub fn user_key_for_seek<'a>(user_key: &[u8], scratch: &'a mut BytesMut) -> Result<&'a [u8]> {
```

Comment:
Why this method is required? It seems this function only copy `user_key` to `scratch`.